<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <!-- configure camera input -->
  <arg name="camera_name" default="/rs_camera_1/color" />
  <arg name="image_topic" default="image_rect" />
  <arg name="info_topic" default="camera_info" />
  <arg name="queue_size" default="1" />

  <!-- apriltag_ros continuous detection node -->
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" clear_params="true" output="screen">
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/$(arg info_topic)" />

    <param name="publish_tag_detections_image" type="bool" value="true" /><!-- default: false -->
    <param name="queue_size" type="int" value="$(arg queue_size)" />

    <!-- load parameters (incl. tag family, tags, etc.) -->
    <rosparam command="load" file="$(find part_detection)/config/april_tag_tracker_settings.yaml"/>
    <rosparam command="load" file="$(find part_detection)/config/april_tags.yaml"/>
  </node>
</launch>